utils/publish-canary.mjs

```diff
@@ -101,7 +101,7 @@ exec('git diff --name-only HEAD HEAD^')
         bump = 'premajor';
       } else {
         // we'll use `next` for pre minors
-        preid = 'next';
+        preid = `${process.env.GITHUB_RUN_NUMBER || 0}-next`;
```

Comment:
Here's the problem. The previous code made release tags like `v7.4.0-next.1`. The `.1` at the end is how many commits exist since the last tag. This doesn't work for us, because we forward merge commits. This means if there's a commit to master that creates a new release tag, the number starts over. The canary job then fails to publish the tag because a `v7.4.0-next.0` already exists. This means for a new `prerelease/minor` to create a canary release, we'll have to have enough commits on the `prerelease/minor` branch to exceed the previous number.

This change instead uses the Github run number to generate a unique release version each time.